src/sql/workbench/browser/modelComponents/declarativeTable.component.ts

```diff
@@ -205,6 +199,16 @@ export default class DeclarativeTableComponent extends ComponentBase implements
 
 	public setProperties(properties: { [key: string]: any; }): void {
 		super.setProperties(properties);
+		if (properties.data) {
+			this.clearContainer();
```

Comment:
So every time we change a property we're going to clear and refresh the grid? Did you test to see if this is visible? I'm worried that we'll get some flickering and/or lose context (such as focus state) by doing this.